pkg/controller/ps/tls_test.go

```diff
@@ -0,0 +1,208 @@
+package ps
+
+import (
+	"context"
+	"path/filepath"
+	"time"
+
+	cm "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+var _ = Describe("Finalizer delete-ssl", Ordered, func() {
+	ctx := context.Background()
+
+	const crName = "delete-ssl-finalizer"
+	const ns = crName
+	crNamespacedName := types.NamespacedName{Name: crName, Namespace: ns}
+
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      crName,
+			Namespace: ns,
+		},
+	}
+
+	BeforeAll(func() {
+		By("Creating the Namespace to perform the tests")
+		err := k8sClient.Create(ctx, namespace)
+		Expect(err).To(Not(HaveOccurred()))
+
+		_, err = envtest.InstallCRDs(cfg, envtest.CRDInstallOptions{
+			Paths: []string{filepath.Join("testdata", "cert-manager.yaml")},
+		})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterAll(func() {
+		time.Sleep(60 * time.Second)
+		By("Deleting the Namespace to perform the tests")
+		_ = k8sClient.Delete(ctx, namespace)
+	})
+
+	Context("delete-ssl finalizer not set", Ordered, func() {
+		cr, err := readDefaultCR(crName, ns)
+		It("should read and create defautl cr.yaml", func() {
+			Expect(err).NotTo(HaveOccurred())
+			Expect(k8sClient.Create(ctx, cr)).Should(Succeed())
+		})
+
+		// This way we simulate cert-manager creating secrets
+		go func() {
+			defer GinkgoRecover()
+
+			// Wait for reconcile run to start creting CM resoureces
+			time.Sleep(10 * time.Second)
```

Comment:
please add a comment about what we're doing here

Reply:
[726ca03](https://github.com/percona/percona-server-mysql-operator/pull/314/commits/726ca034feeb93047a0bf7c6e2b69344dcd6094c)